docs/articles/documentation/extending-testcafe/custom-reporter/creating-reporter-plugin/index.md

```diff
@@ -0,0 +1,197 @@
+```
+
+## Building Reporter
+
+You can build a reporter project by using the `build` Gulp task.
+
+```bash
+gulp build
+```
+
+## Testing Reporter
+
+To test your reporter works well, you can use the `test` Gulp task.
+To get this task to work, you need to provide baseline reports in your format.
+During testing, reports generated by the custom reporter will be compared with these baseline reports.
+
+First, you need to know what data to use in these reports. There are two ways to find this.
+
+* Refer to the test data at `/generator-testcafe-reporter/generators/app/templates/test/data/test-calls.js`, study this JSON file and figure out the meaning of the values in there.
+* Explore an existing reporter, say, the *list* reporter at [https://github.com/DevExpress/testcafe-reporter-list](https://github.com/DevExpress/testcafe-reporter-list). You can either run the `preview` Gulp task to view the reporter output in your terminal, or find the baseline report at `/test/data/report-without-colors`.
+
+Then, you have to compose two baseline reports.
+
+* A regular report. Put it to `/test/data/` and name it `report-without-colors`.
+* A colored report, which is a JSON file that consists of a single string - the report text with additional color information specified using [ANSI escape codes](https://en.wikipedia.org/wiki/ANSI_escape_code). Put this JSON file to `/test/data/` and name it `report-with-colors.json`.
+
+After the baseline reports are ready, you can use the `test` task.
+
+```bash
+gulp test
+```
+
+## Previewing Report
+
+You can preview a report built by your reporter using the `preview` Gulp task.
+
+```bash
+gulp preview
+```
+
+## Using Reporter Development Version
+
+If you are still developing the reporter, but need to test it within TestCafe, there is no need to publish the reporter package to npm.
+You can link the reporter to TestCafe by using the `npm link` command.
+This allows you to work on the reporter project and test it iteratively without having to re-publish everytime you make a change.
+
+Package linking is a two-step process.
+
+1. Run `npm link` from the reporter directory to generate a global reference to the reporter module.
+2. Navigate to the TestCafe directory and run `npm link testcafe-reporter-{reporterName}` to install the reporter module into the TestCafe module.
+
+    Note that `{reporterName}` may not match the reporter directory name. You can find the reporter package name in the `<reporterDirectory>/package.json` file.
+
+**Example**
+
+```bash
+cd MyReporter
+npm link
+cd ../TestCafe
+npm link testcafe-reporter-MyReporter
+```
+
+After that TestCafe will use the reporter version you are currently developing.
+
+For information on how to specify a reporter in tests, see [Using Reporters](/testcafe/documentation/using-testcafe/common-concepts/reporters/#using-reporters).
+
+## Publishing Reporter to NPM
+
+When you finish developing the reporter, you can publish it to npm by using the `publish` Gulp task.
```

Comment:
We need to describe why it should be used instead of regular `npm publish`:
1)It will perform additional checks before publishing to ensure that everything works fine
2)It will built project before publishing